src/middleware.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Custom Koa middleware.
+ */
+
+import BadRequestError from "./errors/BadRequestError";
+
+/**
+ * Handle BadRequestErrors.
+ */
+export async function handleErrors(ctx, next) {
+  try {
+    await next();
+  } catch (error) {
+    if (error instanceof BadRequestError) {
+      ctx.throw(400, error.message);
+    } else {
+      throw error;
```

Comment:
Would be cool if we log errors somewhere, maybe leave a TODO to make that happen